src/serialmonitor/serialMonitor.ts

```diff
@@ -38,256 +30,170 @@ export class SerialMonitor implements vscode.Disposable {
+            this.updatePortListStatus(this.currentPort);
+            return chosen.label;
         }
 
-        if (this._serialPortCtrl) {
-            if (this._currentPort !== this._serialPortCtrl.currentPort) {
-                await this._serialPortCtrl.changePort(this._currentPort);
-            } else if (this._serialPortCtrl.isActive) {
-                vscode.window.showWarningMessage(`Serial monitor is already opened for ${this._currentPort}`);
-                return;
-            }
-        } else {
-            this._serialPortCtrl = new SerialPortCtrl(
-                this._currentPort,
-                this._currentBaudRate,
-                this._currentTimestampFormat,
-                this._bufferedOutputChannel,
-                this._outputChannel.show);
-        }
-
-        if (!this._serialPortCtrl.currentPort) {
-            Logger.traceError("openSerialMonitorError", new Error(`Failed to open serial port ${this._currentPort}`));
-            return;
-        }
-
-        try {
-            await this._serialPortCtrl.open();
-            this.updatePortStatus(true);
-        } catch (error) {
-            Logger.notifyUserWarning("openSerialMonitorError", error,
-                `Failed to open serial port ${this._currentPort} due to error: + ${error.toString()}`);
-        }
+        return undefined;
     }
 
-    public async sendMessageToSerialPort() {
-        if (this._serialPortCtrl && this._serialPortCtrl.isActive) {
-            const text = await vscode.window.showInputBox();
-            try {
-                await this._serialPortCtrl.sendMessage(text);
-            } catch (error) {
-                Logger.notifyUserWarning("sendMessageToSerialPortError", error, constants.messages.FAILED_SEND_SERIALPORT);
-            }
-        } else {
-            Logger.notifyUserWarning("sendMessageToSerialPortError", new Error(constants.messages.SEND_BEFORE_OPEN_SERIALPORT));
-        }
+    public async changeTimestampFormat(): Promise<void> {
+        await vscode.commands.executeCommand("workbench.action.openSettings", "vscode-serial-monitor.timestampFormat");
     }
 
-    public async changeBaudRate() {
+    public async selectBaudRate(): Promise<number | undefined> {
         const rates = SerialMonitor.listBaudRates();
         const chosen = await vscode.window.showQuickPick(rates.map((rate) => rate.toString()));
         if (!chosen) {
-            Logger.warn("No baud rate selected, keeping previous baud rate.");
-            return;
+            return undefined;
         }
         if (!parseInt(chosen, 10)) {
-            Logger.warn("Invalid baud rate, keeping previous baud rate.", { value: chosen });
-            return;
-        }
-        if (!this._serialPortCtrl) {
-            Logger.warn("Serial Monitor has not been started.");
-            return;
+            vscode.window.showWarningMessage(`Invalid baud rate, keeping previous baud rate: ${chosen}`);

```

Comment:
Any particular reason why this was promoted to a notification and the other `!chosen` warning was dropped?

Reply:
Hm, thinking more about it, I don't see a strong reason, changing back.